src/PowerAppsTestEngine/Program.cs

```diff
@@ -125,6 +126,7 @@
             .AddFilter(l => l >= logLevel)
             .AddProvider(new TestLoggerProvider(new FileSystem()));
         })
+        .AddSingleton<ITestEngineEvents>()
```

Comment:
Notice the other scoped services and how they're being added here. You're not actually defining an implementation of your interface, `ITestEngineEvents`. You need to include the actual implementation or this won't work when you run it.